src/main/java/com/example/chat/controller/UserController.java

```diff
@@ -63,26 +66,23 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
                 List<CustomFieldError> errorFields = bindingResult.getFieldErrors().stream()
                         .map(fieldError -> new CustomFieldError(fieldError.getField(), messageSource.getMessage(fieldError.getDefaultMessage(), null, currentLocale)))
                         .collect(Collectors.toList());
-                return ResponseEntity.badRequest().body(errorFields);
+                return ResponseEntity.badRequest().body(ParserToResponseFromCustomFieldError.parseCustomFieldErrors(errorFields));
             } catch (NoSuchMessageException e) {
-                return ResponseEntity.badRequest().body(new ArrayList<>(List.of(
-                        new CustomFieldError("serverError", messageSource.getMessage("server.error", null, currentLocale)))));
+                return ResponseEntity.badRequest().body(new CustomFieldError("serverError", messageSource.getMessage("server.error", null, currentLocale)));
             }
         }
         if (validate.existsByLogin(signUpRequest.getLogin())) {
-            return ResponseEntity.badRequest().body(new ArrayList<>(List.of(
-                    new CustomFieldError("loginDuplicate", messageSource.getMessage("login.duplicate", null, currentLocale)))));
+            return ResponseEntity.badRequest().body(new CustomFieldError("loginDuplicate", messageSource.getMessage("login.duplicate", null, currentLocale)));
         }
         if (validate.existsByEmail(signUpRequest.getEmail())) {
-            return ResponseEntity.badRequest().body(new ArrayList<>(List.of(
-                    new CustomFieldError("emailDuplicate", messageSource.getMessage("email.duplicate", null, currentLocale)))));
+            return ResponseEntity.badRequest().body(new CustomFieldError("emailDuplicate", messageSource.getMessage("email.duplicate", null, currentLocale)));
         }
-        userService.createUser(signUpRequest);
+        userService.createUser(signUpRequest, XOriginatingHost);
         return ResponseEntity.ok(messageSource.getMessage("user.signup.success", null, currentLocale));
     }
 
 
-    @GetMapping("/verification-user-email/{code}")
+    @PutMapping("/verification-user-email/{code}")
```

Comment:
Please consider changing `/verification-user-email/{code}` to `/validate-email/{code}` to make it easier to read